app/waf/src/repositories/common.ts

```diff
@@ -0,0 +1,25 @@
+import {Entity} from '@loopback/repository';
+import {DataObject, Options, DefaultCrudRepository} from '@loopback/repository';
+import uuid = require('uuid');
+
+export class CommonRepository<
+  T extends Entity,
```

Comment:
CommonEntity?

Reply:
Just want to remind that, in your code, you mix-use Entity and CommonEntity: `repositories/common.ts` `controller/schema.ts`. 
Not sure if we use them carefully can avoid some weird issues...

Reply:
Because we will have many-to-many association object, it may not extend from CommonEntity. We may have 2 options: 1) Have the association object extends from CommonEntity. That is the most easy way. 2) Create another repository to serve the association object only. Whatever we select, those code will be remove after we migrate to LB4 implementation of many-to-many relation. I post this notice here, since I hope you not to feel very surprise about the next pr for many-to-many implementation.

Reply:
Have changed the data type to CommonEntity.